Code/Framework/AzCore/AzCore/DOM/DomUtils.h

```diff
@@ -225,8 +225,19 @@ namespace AZ::Dom::Utils
             {
                 return false;
             }
+
             const AZStd::any& opaqueValue = value.GetOpaqueValue();
-            return opaqueValue.is<WrapperType>();
+
+            // Then the original type should be check and not the wrapper type
+            // if the WrapperType is a std::reference_wrapper
+            if constexpr (AZStd::is_reference_wrapper<WrapperType>::value)
+            {
+                return opaqueValue.is<AZStd::remove_reference_t<T>>();
```

Comment:
To fix the failing AttributeDomInteropTests, we need to handle the case where the Dom::Value is actually storing a reference_wrapper to T already.
So we need to check if the opaque is a WrapperType as well like in the else block of this `if constexpr`
```suggestion
                return opaqueValue.is<AZStd::remove_reference_t<T>>() || opaqueValue.is<WrapperType>();
```